content/specs/auth.md

```diff
@@ -0,0 +1,389 @@
+
+The environment a client runs in also impacts the type of redirect (callback) URLs it uses during the Authorization Flow:
+
+- **web clients** include web services and browser apps. Redirect URLs are regular web URLs which open in a browser.
+- **native clients** include some mobile and desktop native clients. Redirect URLs may use platform-specific app callback schemes to open in the app itself.
+
+Authorization Servers may maintain a set of "trusted" clients, identified by `client_id`. Because any client could use unverified client metadata to impersonate a better-known app or brand, Authorization Servers should not display such metadata to end users in the Authorization Interface by default. Trusted clients can have additional metadata shown, such as a readable name (`client_name`), project URI (`client_uri`, which may have a different domain/origin than `client_id`) and logo (`logo_uri`). See the "Security Considerations" section for more details.
+
+Clients which are only using atproto OAuth for account authentication (without authorization to access PDS resources) should request minimal scopes (see “Scopes” section), but still need to implement most of the authorization flow. In particular, they must use the `sub` field in a token response to verify the account identity (this is an atproto-specific detail).
+
+#### Client Metadata
+
+:::note
+The client metadata standard is still a draft and may evolve over time. Our intention is to evolve and align with subsequent drafts and any final standard, while minimizing disruption and breakage with existing implementations.
+:::
+
+Clients must publish a "client metadata" JSON file on the public web. This will be fetched dynamically by Authorization Servers as part of the authorization request (PAR) and at other times during the session lifecycle. The response HTTP status must be 200 (not another 2xx or a redirect), with a JSON object body with the correct `Content-Type` (`application/json`).
+
+Authorization Servers need to fetch client metadata documents from the public web. They should use a hardened HTTP client for these requests (see "OAuth Security Considerations"). Servers may cache client metadata responses, optionally respecting HTTP caching headers (within limits). Minimum and maximum cache TTLs are not currently specified, but should be chosen to ensure that auth token requests predicated on stale confidential client authentication keys (`jwks` or `jwks_uris`) are rejected in a timely manner.
+
+The following fields are relevant for all client types:
+
+- `client_id` (string, required): the `client_id`. Must exactly match the full URL used to fetch the client metadata file itself
```

Comment:
Define exactly here. Case sensitive? Case sensitive path, but not on domain (because domains aren't case sensitive)? Do you want to normalize the URI before comparison?

Reply:
OAuth does not mention normalization. If a `client_id` contains uppercase letters it is fine as long as when fetching that url (following any HTTP spec, eg. regarding the casing of the `Host`) returns the document with a `client_id` with the same casing as initially used.

It the the client's responsibility to provide a `client_id` that matches the metadata. The OAuth Authorization Server will not perform any transformation on the `client_id`.